src/AzureRenderManager/WebApp/Code/Extensions/AzureAdAuthenticationBuilderExtensions.cs

```diff
@@ -47,6 +52,26 @@ public void Configure(string name, OpenIdConnectOptions options)
 
                 // trying to get refresh to work
                 options.Scope.Add("offline_access"); // allow refreshing
+
+                options.Events = new OpenIdConnectEvents
+                {
+                    OnAuthorizationCodeReceived = async ctx =>
+                    {
+                        var request = ctx.HttpContext.Request;
+                        var currentUri = UriHelper.BuildAbsolute(request.Scheme, request.Host, request.PathBase, request.Path);
+                        var credential = new Microsoft.IdentityModel.Clients.ActiveDirectory.ClientCredential(ctx.Options.ClientId, ctx.Options.ClientSecret);
+
+                        var memoryCache = ctx.HttpContext.RequestServices.GetRequiredService<IMemoryCache>();
+                        var cache = new SessionTokenCache(ctx.Principal, memoryCache);
+                        var tokenCache = cache.GetCacheInstance();
+                        var authContext = new AuthenticationContext(ctx.Options.Authority, true, tokenCache);
```

Comment:
what's the reason for us having our own token cache and not using the built-in one?